From 3f9907f93d3983033e176e95f5134a57900a7f6e Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@asymptotic.io>
Date: Thu, 10 Dec 2020 19:06:31 -0500
Subject: [PATCH] build: Use cmake to look up abseil dependency

This should be much more robust than looking up the library directly.

Fixes: https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/issues/4

Reference:
https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/commit/3f9907f93d3983033e176e95f5134a57900a7f6e
https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/commit/ce1a78887a1a0dfed3ebdb4de964d7afefebe896
https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/commit/8ce8bebb7d852491326af1075ead74e757ae4611
---
 meson.build | 27 +++++++++++----------------
 1 file changed, 11 insertions(+), 16 deletions(-)

diff --git a/meson.build b/meson.build
index de8cd75..96d84d7 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
-project('webrtc-audio-processing-1', 'c', 'cpp',
-  version : '1.0',
+project('webrtc-audio-processing', 'c', 'cpp',
+  version : '1.1',
   meson_version : '>= 0.54',
   default_options : [ 'warning_level=1',
                       'buildtype=debugoptimized',
@@ -27,7 +27,7 @@ ac_minor_version = minor_version
 ac_version = ac_major_version + '.' + ac_minor_version
 ac_project_name = 'webrtc-audio-coding-' + ac_major_version
 
-include_subdir = meson.project_name()
+include_subdir = apm_project_name
 
 cc = meson.get_compiler('c')
 cpp = meson.get_compiler('cpp')
@@ -40,19 +40,14 @@ os_deps = []
 have_posix = false
 have_win = false
 
-absl_dep = [
-  cpp.find_library('absl_base'),
-  cpp.find_library('absl_bad_optional_access'),
-  cpp.find_library('absl_flags_internal'),
-  cpp.find_library('absl_flags_marshalling'),
-  cpp.find_library('absl_flags_parse'),
-  cpp.find_library('absl_flags_registry'),
-  cpp.find_library('absl_flags_usage_internal'),
-  cpp.find_library('absl_raw_logging_internal'),
-  cpp.find_library('absl_strings'),
-  cpp.find_library('absl_synchronization'),
-  cpp.find_library('absl_throw_delegate'),
-]
+absl_dep = dependency('absl', method : 'cmake',
+  modules : [
+    'absl::base',
+    'absl::flags_parse',
+    'absl::strings',
+    'absl::synchronization',
+  ]
+)
 
 if ['darwin', 'ios'].contains(host_system)
   os_cflags = ['-DWEBRTC_MAC']
-- 
2.27.0

